File: pgm/ABC_Java/ABC_298_A.java
import java.io.*;
// import java.math.*;
import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        reader.readLine();
        final String[] S = reader.readLine().split ("");
        final String GOOD = "o";
        final String FAILING = "x";
        final String Yes = "Yes";
        final String No = "No";
        System.out.println (
            Arrays.stream (S).anyMatch (s -> GOOD.equals (s)) &&
                    Arrays.stream (S).noneMatch (s -> FAILING.equals (s))
                ? Yes
                : No);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_298_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.Scanner
// import java.util.StringTokenizer

// import kotlin.math.*

const val good: String = "o"
const val failing: String = "x"
const val Yes: String = "Yes"
const val No: String = "No"
const val blank: String = ""

fun main() {
    readln()
    val S: Array<String> = readln().split(blank).toTypedArray()
    if (S.any { it == good } && S.none { it == failing }) {
        println(Yes)
    } else {
        println(No)
    }
    kotlin.system.exitProcess(0)
}